components/Chat/Message/Message.tsx

```diff
@@ -133,9 +145,55 @@ const MessageSenderAvatar = ({ message }: { message: MessageToDisplay }) => {
   );
 };
 
-function ChatMessage({ message, colorScheme, isGroup, isFrame }: Props) {
+const ChatMessage = ({
+  account,
+  message,
+  colorScheme,
+  isGroup,
+  isFrame,
+  showTime,
+  toggleTime,
+}: Props & { showTime: boolean; toggleTime: () => void }) => {
   const styles = useStyles();
 
+  const messageDate = useMemo(
+    () => getRelativeDate(message.sent),
+    [message.sent]
+  );
+  const messageTime = useMemo(
+    () => getLocalizedTime(message.sent),
+    [message.sent]
+  );
+
+  // Reanimated shared values for height, translateY, and opacity
+  const height = useSharedValue(0);
+  const translateY = useSharedValue(20);
+  const opacity = useSharedValue(0);
+
+  const animatedStyle = useAnimatedStyle(() => {
+    return {
+      minHeight: height.value, // Use minHeight for more stable height animations
+      transform: [{ translateY: translateY.value }],
+      opacity: opacity.value,
+    };
+  });
+
+  // Handle animation based on showTime prop
+  React.useEffect(() => {
```

Comment:
this useEffect will be triggered when the component renders then when the props change

when the component renders showTime will be false so the "hide time" animation will trigger but it's not necessary since it's already hidden

not sure if it's really a perf issue but we can  prevent that: add a useRef for the last showtime value starting with `false` and only trigger the show animation if showtime goes from false to true or from true to false